library/net.c

```diff
@@ -330,12 +330,19 @@ static int dnet_io_req_queue(struct dnet_net_state *st, struct dnet_io_req *orig
 
 void dnet_io_req_free(struct dnet_io_req *r)
 {
+	common_request_free(r->common_req);
```

Comment:
`common_req` is C++ struct (allocated via `new`) and has virtual destructor. It's a base for all requests from `protocol.hpp` which are also C++ structs since they have some non-POD fields. `common_request_free` must be implemented on C++ call operator `delete`.